apps/autoscheduler/LoopNest.cpp

```diff
@@ -0,0 +1,1861 @@
+                const auto &task_bounds = consumer_task_site->get_bounds(e->producer);
+                const auto &producer_compute_bounds = producer_compute_site->get_bounds(e->producer);
+                const auto &producer_store_bounds = producer_store_site->get_bounds(e->producer);
+
+                // Compute memory footprints in terms of the
+                // number of contiguous lines, and the number of
+                // bytes.
+                int64_t footprint = e->producer->bytes_per_point;
+                int64_t compute_footprint = footprint;
+                int64_t store_footprint = footprint;
+                int64_t task_footprint = footprint;
+                int64_t line_footprint = 1;
+                int64_t compute_line_footprint = 1;
+                int64_t store_line_footprint = 1;
+                int64_t task_line_footprint = 1;
+
+                if (e->producer->is_input) {
+                    // This node represents an input. Its sites
+                    // should be at the root level.
+                    internal_assert(producer_store_site->is_root());
+                    internal_assert(producer_compute_site->is_root());
+                }
+
+                if (innermost) {
+
+                    // Grab the Jacobians that describe the memory dependence
+                    for (const auto &jac : jacobians) {
+                        if (jac.second != e->producer) continue;
+                        double n = jac.first.count();
+
+                        // Classify them to figure out what's going on in the vector dimension.
+                        bool vector_broadcast = true;
+                        bool dense_vector_load = true;
+                        bool stride_2_vector_load = true;
+                        bool stride_3_vector_load = true;
+                        bool stride_4_vector_load = true;
+                        int producer_innermost_dim =
+                            (e->producer->is_input ? 0 :  // Assume default storage layout for inputs
+                                 !producer_has_been_scheduled ? -1 :
+                                                                site.produce->vector_dim);
+                        if (vectorized_loop_index >= 0) {
+                            if (!producer_has_been_scheduled) {
+                                // Operate optimistically and just
+                                // see if *any* dimension of the
+                                // producer would make for a good
+                                // load.
+                                int count[5] = {0, 0, 0, 0, 0};
+                                for (int i = 0; i < e->producer->dimensions; i++) {
+                                    auto stride = jac.first(i, vectorized_loop_index);
+                                    // stride is a rational. Check to see if it's a small integer.
+                                    if (stride == 0)
```

Comment:
Nit: always enclose if statements in {braces}, even single-line ones

Reply:
done